lib/sles4sap/sdaf_library.pm

```diff
@@ -513,14 +720,124 @@ sub prepare_sdaf_repo {
+
+=cut
+
+sub generate_resource_group_name {
+    my (%args) = @_;
+    my @supported_types = ('workload_zone', 'sap_system', 'library', 'deployer');
+    croak "Unsupported deployment type: $args{deployment_type}\nCurrently supported ones are: @supported_types" unless
+      grep(/^$args{deployment_type}$/, @supported_types);
+    my $job_id = get_current_job_id();
+    my $resource_group = join('-', 'SDAF', 'OpenQA', $args{deployment_type}, $job_id);
+
+    return $resource_group;
+}
+
+=head2 resource_group_exists
+
+    resource_group_exists($resource_group);
+
+B<$resource_group>: Resource group name to check
+
+Checks if resource group exists. Function accepts only full resource name.
+Croaks if command does not return true/false value.
+
+=cut
+
+sub resource_group_exists {
+    my ($resource_group) = @_;
+    croak 'Madatory positional argument "$resource_group" not defined.' unless $resource_group;
+
+    my $cmd_out = script_output("az group exists -n $resource_group");
+    die "Command 'az group exists -n $resource_group' failed.\nCommand returned: $cmd_out" unless grep /false|true/, $cmd_out;
+    return 0 if $cmd_out eq 'false';
+    return 1 if $cmd_out eq 'true';
+}
+
+=head2 sdaf_execute_remover
+
+    sdaf_execute_remover(deployment_type=>$deployment_type);
 
-    cleanup_sdaf_files();
+B<$deployment_type>: Type of the deployment (workload_zone, sap_system)
 
-Cleans up all SDAF deployment files belonging to the running test.
+Uses remover.sh script which is part of the SDAF project. This script can be used only on workload zone or sap system.
+Control plane and library have separate removal script, but are currently part of permanent setup and should not be destroyed.
+Returns RC to allow additional cleanup tasks required even after script failure.
+L<https://learn.microsoft.com/en-us/azure/sap/automation/bash/remover>
 
 =cut
 
-sub cleanup_sdaf_files {
+sub sdaf_execute_remover {
+    my (%args) = @_;
+    croak 'Missing mandatory positional argument "$deployment_type"' unless $args{deployment_type};
+    croak 'This function can be used only on sap system and workload zone removal' unless
+      grep /^$args{deployment_type}$/, ('sap_system', 'workload_zone');
+
+    # SDAF remover.sh uses term 'sap_landscape' for 'workload_zone'.
+    my $type_parameter = $args{deployment_type} eq 'workload_zone' ? 'sap_landscape' : $args{deployment_type};
+
+    my $tfvars_file;
+    $tfvars_file = get_os_variable('sap_system_parameter_file') if $args{deployment_type} eq 'sap_system';
+    $tfvars_file = get_os_variable('workload_zone_parameter_file') if $args{deployment_type} eq 'workload_zone';
+    die 'Function failed to retrieve tfvars file via OS variable.' unless $tfvars_file;
+
+    my ($tfvars_filename, $tfvars_path) = fileparse($tfvars_file);
+    my $remover_cmd = join(' ',
+        deployment_dir() . '/sap-automation/deploy/scripts/remover.sh',
```

Comment:
other places are composing this path in a different way
https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19048/files#diff-a7b59ba45f396aa9b361d43f76fb4f6669a4b42c5d0e1dc952bfa526ec508934R632

Reply:
Oh I see. I switched the other one to deployment_dir() as well.